slasher/db/proposer_slashings.go

```diff
@@ -0,0 +1,233 @@
+package db
+
+import (
+	"bytes"
+
+	"github.com/prysmaticlabs/go-ssz"
+
+	"github.com/boltdb/bolt"
+	"github.com/gogo/protobuf/proto"
+	"github.com/pkg/errors"
+	ethpb "github.com/prysmaticlabs/ethereumapis/eth/v1alpha1"
+)
+
+// SlashingStatus enum like structure.
+type SlashingStatus uint8
+
+const (
+	// Unknown default status in case it is not set
+	Unknown = iota
+	// Active slashing proof hasn't been included yet.
+	Active
+	// Included slashing proof that has been included in a block.
+	Included
+	// Reverted slashing proof that has been reverted and therefore is relevant again.
+	Reverted //relevant again
+)
+
+func (status SlashingStatus) String() string {
+	names := [...]string{
+		"Unknown",
+		"Active",
+		"Included",
+		"Reverted"}
+
+	if status < Active || status > Reverted {
+		return "Unknown"
+	}
+	// return the name of a SlashingStatus
+	// constant from the names array
+	// above.
+	return names[status]
+}
+
+// SlashingType enum like type of slashing proof.
+type SlashingType uint8
+
+const (
+	// Proposal enum value.
+	Proposal = iota
+	// Attestation enum value.
+	Attestation
+)
+
+func (status SlashingType) String() string {
+	names := [...]string{
+		"Proposal",
+		"Attestation"}
+
+	if status < Active || status > Reverted {
+		return "Unknown"
+	}
+	// return the name of a SlashingType
+	// constant from the names array
+	// above.
+	return names[status]
+}
+
+func createProposerSlashing(enc []byte) (*ethpb.ProposerSlashing, error) {
+	protoSlashing := &ethpb.ProposerSlashing{}
+
+	err := proto.Unmarshal(enc, protoSlashing)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to unmarshal encoding")
+	}
+	return protoSlashing, nil
+}
+
+func toProposerSlashings(encoded [][]byte) ([]*ethpb.ProposerSlashing, error) {
+	var proposerSlashings []*ethpb.ProposerSlashing
+	for _, enc := range encoded {
+		ps, err := createProposerSlashing(enc)
+		if err != nil {
+			return nil, err
+		}
+		proposerSlashings = append(proposerSlashings, ps)
+	}
+	return proposerSlashings, nil
+}
+
+// ProposerSlashings accepts a status and returns all slashings with this status.
+// returns empty proposer slashing slice if no slashing has been found with this status.
+func (db *Store) ProposerSlashings(status SlashingStatus) ([]*ethpb.ProposerSlashing, error) {
+	encoded := make([][]byte, 0)
+	err := db.view(func(tx *bolt.Tx) error {
+		c := tx.Bucket(slashingBucket).Cursor()
+		prefix := encodeStatusType(status, SlashingType(Proposal))
+		for k, v := c.Seek(prefix); k != nil && bytes.HasPrefix(k, prefix); k, v = c.Next() {
+			encoded = append(encoded, v)
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
+	return toProposerSlashings(encoded)
+}
+
+// ProposalSlashingsByStatus returns all the proposal slashing proofs with a certain status.
+func (db *Store) ProposalSlashingsByStatus(status SlashingStatus) ([]*ethpb.ProposerSlashing, error) {
```

Comment:
again this is the same exact method as the above one.

Reply:
thanks, removed